src/librustc_resolve/diagnostics.rs

```diff
@@ -425,6 +521,55 @@ match 0 {
 ```
 "##,
 
+E0419: r##"
+An unknown enum variant, struct or const was used. Example of
+erroneous code:
+
+```
+match 0 {
+    Something::Foo => {} // error: unresolved enum variant, struct
+                         //        or const `Foo`
+}
+```
+
+Please verify you didn't misspell it or the enum variant, struct
+or const was well declared and/or imported. Example:
```

Comment:
Please verify you didn't misspell it and the enum variant, struct or const has been declared and imported into scope.